charmhelpers/contrib/hahelpers/cluster.py

```diff
@@ -361,3 +365,24 @@ def canonical_url(configs, vip_setting='vip'):
     else:
         addr = unit_get('private-address')
     return '%s://%s' % (scheme, addr)
+
+
+def distributed_wait(modulo=None, wait=None):
+    ''' Distribute operations by waiting based on modulo_distribution
+
+    If modulo and or wait are not set, check config_get for those values.
+
+    :side effect: Calls config_get()
+    :side effect: Calls log()
+    :side effect: Calls status_set()
+    :side effect: Calls time.sleep()
+    '''
+    if modulo is None:
+        modulo = config_get('modulo-nodes')
+    if wait is None:
+        wait = config_get('known-wait')
+    calculated_wait = modulo_distribution(modulo=modulo, wait=wait)
+    msg = "Waiting {} seconds for operation ...".format(calculated_wait)
```

Comment:
I find operation to be a vague word from a user perspective here. The end user will get more value out of it if this says what the operation is rather than just operation (e.g. "Waiting n seconds to restart" or something). What about making this customizable?

Reply:
Sounds good.